src/accelerate/accelerator.py

```diff
@@ -862,6 +862,52 @@ def no_sync(self, model):
         with context():
             yield
 
+    @staticmethod
+    @contextmanager
+    def ddp_trigger_sync_in_bwd(model_ddp):
+        """Trigger the sync of the gradients in the next backward pass of DDP model.
+
+        If `model` is not in DDP, this context manager does nothing
+
+        Args:
+            model_ddp (`torch.nn.parallel.DistributedDataParallel`):
+                PyTorch Module wrapped in `DistributedDataParallel`
+
+        Example:
+
+        ```python
+        >>> from accelerate import Accelerator
+
+        >>> accelerator = Accelerator()
+        >>> dataloader, model, optimizer = accelerator.prepare(dataloader, model, optimizer)
+
+        >>> with accelerator.no_sync():
+        ...     loss_a = loss_func(model(input_a)) # first forward pass
+        ...     loss_b = loss_func(model(input_b)) # second forward pass
+        >>> accelerator.backward(loss_a) # No synchronization across processes, only accumulate gradients
+        >>> with accelerator.ddp_trigger_sync_in_bwd(model):
+        ...     accelerator.backward(loss_b) # Synchronization across all processes
+        >>> optimizer.step()
+        >>> optimizer.zero_grad()
+        ```
+        """
+        if not isinstance(model_ddp, torch.nn.parallel.DistributedDataParallel):
+            yield
+            return
```

Comment:
Not sure you can put a `return` statement since there is a `yield`. You might need to do an else branch.

Reply:
I just tested it and it works.
But if you prefer the `if.. else` I can switch to that

Reply:
As long as Python is happy, I'm happy.